src/plugins/main.js

```diff
@@ -114,6 +127,18 @@ export default async (context) => {
   app.i18n.beforeLanguageSwitch = <%= options.beforeLanguageSwitch %>
   app.i18n.onLanguageSwitched = <%= options.onLanguageSwitched %>
   app.i18n.setLocaleCookie = setLocaleCookie
+  app.i18n.getLocaleCookie = getLocaleCookie
+
+  const { loadLanguageAsync } = require('./utils')
+  app.i18n.setLocale = async (locale) => {
+    await loadLanguageAsync(context, locale)
+    app.i18n.locale = locale
+
+    if (useCookie) {
+      setLocaleCookie(locale)
+    }
+  }
```

Comment:
We should also call `syncVuex`.

There is also an issue with this API when using other than `no-prefix` strategies. It will change locale but route URL will stay unchanged (with previous locale prefix). I'm not sure what's the best way to handle that - two options I see are: call nuxt's `redirect` or replace route usign `VueRouter.replace()`. Both can be destructive but I guess that should be OK... I can handle that myself later.